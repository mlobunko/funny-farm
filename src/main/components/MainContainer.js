import React from 'react';

import ErrorBoundary from './ErrorBoundary';
import MainComponent from './MainComponent';

export class MainContainer extends React.Component {
  state = {
    isLandscape: true
  };
  landscapeOn = () => {
    this.setState({
      isLandscape: true
    });
  };
  landscapeOff = () => {
    this.setState({
      isLandscape: false
    });
  };
  checkLandscape = () =>
    document.documentElement.clientWidth > document.documentElement.clientHeight
      ? this.landscapeOn()
      : this.landscapeOff();

  componentDidMount = () => {
    this.checkLandscape();
    window.addEventListener('resize', this.checkLandscape);
  };
  render() {
    return (
      <ErrorBoundary>
        <MainComponent isLandscape={this.state.isLandscape} />
      </ErrorBoundary>
    );
  }
}

export default MainContainer;
