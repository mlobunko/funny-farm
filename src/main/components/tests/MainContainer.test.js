import React from 'react';
import { shallow } from 'enzyme';

import { MainContainer } from '../MainContainer';

describe('render', () => {
  it('renders correctly, isLandscape true', () => {
    const wrapper = shallow(<MainContainer />);
    wrapper.setState({ isLandscape: true });
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correctly, isLandscape false', () => {
    const wrapper = shallow(<MainContainer />);
    wrapper.setState({ isLandscape: false });
    expect(wrapper).toMatchSnapshot();
  });
});

describe('landscapeOn', () => {
  it('set isLandscape to true', () => {
    const wrapper = shallow(<MainContainer />);
    wrapper.setState({ isLandscape: false });
    wrapper.instance().landscapeOn();
    expect(wrapper.state('isLandscape')).toEqual(true);
  });
});

describe('landscapseOff', () => {
  it('set isLandscape to false', () => {
    const wrapper = shallow(<MainContainer />);
    wrapper.setState({ isLandscape: true });
    wrapper.instance().landscapeOff();
    expect(wrapper.state('isLandscape')).toEqual(false);
  });
});

describe('checkLandscape', () => {
  const clientHeight = 200;
  const clientWidth = 100;

  Object.defineProperty(document, 'documentElement', {
    value: { clientHeight, clientWidth }
  });

  it('call landscapeOn', () => {
    const wrapper = shallow(<MainContainer />);
    wrapper.instance().landscapeOn = jest.fn();
    wrapper.instance().landscapeOff = jest.fn();
    wrapper.instance().checkLandscape();
    expect(wrapper.instance().landscapeOff).toHaveBeenCalledTimes(1);
    expect(wrapper.instance().landscapeOn).toHaveBeenCalledTimes(0);
  });

  it('call landscapeOff', () => {
    document.documentElement.clientHeight = 100;
    document.documentElement.clientWidth = 200;
    const wrapper = shallow(<MainContainer />);
    wrapper.instance().landscapeOn = jest.fn();
    wrapper.instance().landscapeOff = jest.fn();
    wrapper.instance().checkLandscape();
    expect(wrapper.instance().landscapeOff).toHaveBeenCalledTimes(0);
    expect(wrapper.instance().landscapeOn).toHaveBeenCalledTimes(1);
  });
});

describe('componentDidMount', () => {
  Object.defineProperty(window, 'addEventListener', {
    value: jest.fn()
  });

  beforeEach(() => {
    jest.resetAllMocks();
  });

  it('call checkLandscape', () => {
    const wrapper = shallow(<MainContainer />);
    wrapper.instance().checkLandscape = jest.fn();
    wrapper.instance().componentDidMount();
    expect(wrapper.instance().checkLandscape).toHaveBeenCalledTimes(1);
  });

  it('call addEventListener', () => {
    const wrapper = shallow(<MainContainer />);
    expect(window.addEventListener).toHaveBeenCalledWith(
      'resize',
      wrapper.instance().checkLandscape
    );
  });
});
