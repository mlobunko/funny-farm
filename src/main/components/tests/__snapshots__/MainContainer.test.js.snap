// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render renders correctly, isLandscape false 1`] = `
<ErrorBoundary>
  <Component
    isLandscape={false}
  />
</ErrorBoundary>
`;

exports[`render renders correctly, isLandscape true 1`] = `
<ErrorBoundary>
  <Component
    isLandscape={true}
  />
</ErrorBoundary>
`;
